.github/workflows/testing.yaml

```diff
@@ -0,0 +1,27 @@
+name: PR Test Workflow
+
+on:
+  pull_request:
+    branches:
+      - "*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0
+
+      - name: Restore dependencies
+        run: dotnet restore
+
+      - name: Build and Test
+        run: dotnet test --no-restore --verbosity normal /p:CollectCoverage=true

```

Comment:
We may not need to collect coverage during tests, removing the coverage recollection in this step should make the pipe execution faster